xilem/src/view/worker.rs

```diff
@@ -58,88 +59,80 @@ where
 ///
 /// This is [`worker`] without the capturing rules.
 /// See `worker` for full documentation.
-pub fn worker_raw<M, V, F, H, State, Action, Fut>(
-    value: V,
+pub fn worker_raw<M, V, S, F, H, State, Action, Fut>(
     init_future: F,
+    store_sender: S,
     on_response: H,
-) -> Worker<F, H, M, V>
+) -> Worker<F, H, M, S, V>
 where
     // TODO(DJMcNab): Accept app_state here
     F: Fn(MessageProxy<M>, UnboundedReceiver<V>) -> Fut,
     Fut: Future<Output = ()> + Send + 'static,
+    S: Fn(&mut State, UnboundedSender<V>),
     H: Fn(&mut State, M) -> Action + 'static,
     M: AnyMessage + 'static,
 {
     Worker {
-        value,
         init_future,
         on_response,
+        store_sender,
         message: PhantomData,
     }
 }
 
-pub struct Worker<F, H, M, V> {
+pub struct Worker<F, H, M, S, V> {
     init_future: F,
-    value: V,
+    store_sender: S,
     on_response: H,
-    message: PhantomData<fn() -> M>,
+    message: PhantomData<fn(M, V)>,
 }
 
-#[doc(hidden)] // Implementation detail, public because of trait visibility rules
-pub struct WorkerState<V> {
-    handle: JoinHandle<()>,
-    sender: UnboundedSender<V>,
-}
-
-impl<F, H, M, V> ViewMarker for Worker<F, H, M, V> {}
+impl<F, H, M, S, V> ViewMarker for Worker<F, H, M, S, V> {}
 
-impl<State, Action, V, F, H, M, Fut> View<State, Action, ViewCtx> for Worker<F, H, M, V>
+impl<State, Action, F, H, M, Fut, S, V> View<State, Action, ViewCtx> for Worker<F, H, M, S, V>
 where
     F: Fn(MessageProxy<M>, UnboundedReceiver<V>) -> Fut + 'static,
-    V: Send + PartialEq + Clone + 'static,
+    V: Send + 'static,
     Fut: Future<Output = ()> + Send + 'static,
+    S: Fn(&mut State, UnboundedSender<V>) + 'static,
     H: Fn(&mut State, M) -> Action + 'static,
     M: AnyMessage + 'static,
 {
     type Element = NoElement;
 
-    type ViewState = WorkerState<V>;
+    type ViewState = JoinHandle<()>;
```

Comment:
There are reasonable arguments for making this still a struct, but I agree that we don't have any strong guidelines here.